akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala

```diff
@@ -183,5 +183,5 @@ final class Tcp(system: ExtendedActorSystem) extends akka.actor.Extension {
    * It represents a prospective TCP client connection to the given endpoint.
    */
   def outgoingConnection(host: String, port: Int): Flow[ByteString, ByteString, Future[OutgoingConnection]] =
-    outgoingConnection(new InetSocketAddress(host, port))
+    outgoingConnection(InetSocketAddress.createUnresolved(host, port))
```

Comment:
Actually I use `Http().outgoingConnection(remoteAddress = unresolved)` so this should hopefully not break something since akka will resolve it, then